src/WikiService.Business/Commands/Article/Create/CreateArticleHandler.cs

```diff
@@ -16,21 +18,34 @@ public class CreateArticleHandler : IRequestHandler<CreateArticleRequest, Guid?>
   {
     private readonly IHttpContextAccessor _httpContextAccessor;
     private readonly IDataProvider _provider;
-    private readonly IMemoryCache _cache;
-
-    public CreateArticleHandler(
-      IHttpContextAccessor httpContextAccessor,
-      IDataProvider provider,
-      IMemoryCache cache)
+    private readonly IMemoryCache _cache;
+
+    private async Task<Guid?> CreateAsync(DbArticle dbArticle, CancellationToken ct)
     {
-      _httpContextAccessor = httpContextAccessor;
-      _provider = provider;
-      _cache = cache;
+      if (dbArticle is null)
+      {
+        return null;
+      }
+
+      int? position = await _provider.Articles
+        .Where(x => x.RubricId == dbArticle.RubricId)
+        .MaxAsync(x => (int?)x.Position);
+      dbArticle.Position = !position.HasValue ? 1 : position.Value + 1;

```

Comment:
may be
int position = await _provider.Articles
        .Where(x => x.RubricId == dbArticle.RubricId)
        .CountAsync();
dbArticle.Position = position + 1;

and the same with rubrics

Reply:
can't we get position from create request?

Reply:
new article/rubric stands at the end of line by default; we don't receive it in request

Reply:
done